src/pages/ContentScripts/features/fast-pr/githubService.ts

```diff
@@ -0,0 +1,282 @@
+import * as url from './githubUrl';
+import { handleMessage } from './handleMessage';
+import type { FormInstance } from 'antd/lib/form';
+import { getGithubToken } from '../../../../helpers/github-token';
+import i18n from '../../../../helpers/i18n';
+export const PR_TITLE = (file: string) => `docs: Update ${file}`;
+export const PR_CONTENT = (file: string) => `Update ${file} by [FastPR](https://github.com/hypertrons/hypertrons-crx).`;
+const t = i18n.t;
+const generateBranchName = () => `fastPR-${new Date().toISOString().replace(/:/g, '-').replace(/\..+/, '')}`;
+const COMMIT_MESSAGE = (branch: string) => `docs: ${branch}`;
+// Check if the repo has been forked
+const checkRepositoryPermission = async (repoName: string, githubToken: string) => {
+  const headers = {
+    Authorization: `Bearer ${githubToken}`,
+    Accept: 'application/vnd.github.v3+json',
+  };
+
+  // Step 1: Obtain repo information and check permissions
+  const repoResponse = await fetch(url.GET_REPO_INFO(repoName), { headers });
+  if (!repoResponse.ok) {
+    return {
+      success: false,
+      message: t('error_fetch_repo_info', { status: repoResponse.status }),
+    };
+  }
+
+  const repoData = await repoResponse.json();
+
+  // Check if the user has write permission
+  if (repoData.permissions && repoData.permissions.push) {
+    return {
+      success: true,
+      permission: true,
+    };
+  } else {
+    return {
+      success: true,
+      permission: false,
+    };
+  }
+};
+const getOrCreateFork = async (repoName: string, githubToken: string) => {
```

Comment:
I see that you've encapsulated a lot of the GitHub v3 API. Have you considered using the [Octokit](https://github.com/octokit/octokit.js) library for making requests? This way, you wouldn't need to maintain endpoints, parameters, and return values. The library is synchronized with the GitHub API and comes with comprehensive type declarations.

Reply:
> I see that you've encapsulated a lot of the GitHub v3 API. Have you considered using the [Octokit](https://github.com/octokit/octokit.js) library for making requests? This way, you wouldn't need to maintain endpoints, parameters, and return values. The library is synchronized with the GitHub API and comes with comprehensive type declarations.

May I ask if there are any libraries similar to Octokit for the Gitee API？

Reply:
I did not know and find any lib like Octokit for Gitee, we may need to develop ourselves.